pyproject.toml

```diff
@@ -2,14 +2,115 @@
 # Minimum requirements for the build system to execute.
 # See https://github.com/scipy/scipy/pull/12940 for the AIX issue.
 requires = [
-    "setuptools>=51.0.0",
+    "meson-python",
     "wheel",
     "Cython>=0.29.32,<3",  # Note: sync with setup.py, environment.yml and asv.conf.json
     "oldest-supported-numpy>=2022.8.16"
 ]
-# uncomment to enable pep517 after versioneer problem is fixed.
-# https://github.com/python-versioneer/python-versioneer/issues/193
-# build-backend = "setuptools.build_meta"
+build-backend = "mesonpy"
+
+[project]
```

Comment:
Could you move those to a precursor PR? And remove those from `setup.cfg`

Reply:
Also, bump `setuptools` to 61.0.0

Reply:
I started to work on this, but on second thought this is not possible, since we can't do a PEP 517 build with setuptools yet. #39415.

This can't go after, either, since CI fails without it. :(

Reply:
Isn't meson-python going to completely replace setuptools anyway? Not sure we need to worry about setuptools for this PR, though would be important in the timespan that we use both setuptools + meson-python

Reply:
I was talking about not being able to split moving config options from setup.cfg -> pyproject.toml out of this PR. Sorry for the unclear comment.

Reply:
I think writing metadata in pyproject.toml is PEP 621, not related to PEP 517 build. 

IMO, moving config and metadata from `setup.cfg` to `pyproject.toml` is a separate issue, and in this case, it's a prerequisite for using meson.